chain/epoch_manager/src/lib.rs

```diff
@@ -38,7 +38,6 @@ mod reward_calculator;
 #[cfg(feature = "protocol_feature_chunk_only_producers")]
 mod shard_assignment;
 pub mod test_utils;
-mod tests;
```

Comment:
Curious, dose `#[cfg(test)]` work here? If it does, that would be better than introducing the nested test module.

Reply:
Am I right that your proposal is trying to move that tests module from https://github.com/near/nearcore/pull/6390/files#diff-d53b6e2ee18a5c4ef85c47a6f020198cfb91d4e32d96ec2b799e412effc3c606R1 and place it near thttps://github.com/near/nearcore/blob/622e4439097a36856d1bf9839f0dac9b071282b8/chain/epoch_manager/src/lib.rs#L1498 ?

Reply:
Oh, this is a bit confusing, let me elaborate!

So, in the current state of the PR, you have this `#[cfg(test)]` module, but it feels that the module *itself* is already inside test-only code, so `#[cfg(test)]` should be pulled up, and guard the whole thing.

It seems like the `mod tests;` declaration in the `src/lib.rs` is the place where we need to add a cfg.

Taking a second look here, it seems that, in the current state, the code is dead because we miss `mod tests;` altogether.

This is what I think should work:

```diff
diff --git a/chain/epoch_manager/src/lib.rs b/chain/epoch_manager/src/lib.rs
index eeb299d00..5e70221ce 100644
--- a/chain/epoch_manager/src/lib.rs
+++ b/chain/epoch_manager/src/lib.rs
@@ -41,6 +41,9 @@ pub mod test_utils;
 mod types;
 mod validator_selection;
 
+#[cfg(test)]
+mod tests;
+
 const EPOCH_CACHE_SIZE: usize = if cfg!(feature = "no_cache") { 1 } else { 50 };
 const BLOCK_CACHE_SIZE: usize = if cfg!(feature = "no_cache") { 5 } else { 1000 }; // TODO(#5080): fix this
 const AGGREGATOR_SAVE_PERIOD: u64 = 1000;
diff --git a/chain/epoch_manager/src/tests/random_epochs.rs b/chain/epoch_manager/src/tests/random_epochs.rs
index 9b9a4f5ff..7b5dfe303 100644
--- a/chain/epoch_manager/src/tests/random_epochs.rs
+++ b/chain/epoch_manager/src/tests/random_epochs.rs
@@ -1,5 +1,5 @@
-#[cfg(test)]
-mod test {
+// #[cfg(test)]
+// mod test {
     use rand::prelude::StdRng;
     use rand::{Rng, SeedableRng};
     use std::collections::{BTreeMap, BTreeSet, HashMap};
@@ -406,4 +406,4 @@ mod test {
             .map(|stake| stake.account_and_stake())
             .collect::<HashMap<_, _>>()
     }
-}
+// }
```

Reply:
Yes, I really missed an idea about marking the whole module as tests module on the top level.
Thanks.